snippets/survival_analysis/Coxph_model/Coxph_survival_regression.Rmd

```diff
@@ -8,61 +8,90 @@ author: "Jana Fischer"
 knitr::opts_chunk$set(echo = TRUE)
 ```
 
-# Read in data
+# Read in survival data
 ```{r}
 library(data.table)
+library(survival)
+library(ggplot2)
 
-surv_dat = fread('/home/jana/Github/bbsnippets/data/survival_example_data.csv',header = T)
+surv_dat = fread('~/Github/bbsnippets/data/survival_example_data.csv',header = T)
 head(surv_dat)
 
-# Column description:
-## PID: patient identifier
-## grade: tumor pathology grade
-## OSmonth: overall suvival from time of diagnosis [months]
-## DFSmonth: disease free survival from time of diagnosis [months]
-## Patientstatus: life/death status 
-### "death by primary disease" = death due to disease
-### "death" = disease unrelated death
-### "alive" = alive at given survival time
-### "alive w metastases" = alive with metasatses at given survival time
-
 ```
 
+Column description:  
+ PID: patient identifier  
+ grade: tumor pathology grade  
+ OSmonth: overall suvival from time of diagnosis [months]  
+ DFSmonth: disease free survival from time of diagnosis [months]  
+ Patientstatus: life/death status   
+  "death by primary disease" = death due to disease  
+  "death" = disease unrelated death  
+  "alive" = alive at given survival time  
+  "alive w metastases" = alive with metasatses at given survival time  
+
+
 # Prepare censoring information
-```{r}
 
-#Censoring for overall survival (1 = event (death due to disease), 0 = censored)
+Survival data is censored when the event has not yet occurred for a patient
+at a given time or the patient is lost to follow-up during the study.
+A more detailed explanation can be found here: 
+http://www.sthda.com/english/wiki/survival-analysis-basics#censoring
+
+```{r}
+# Censoring for overall survival (1 = event (death due to disease), 0 = censored)
 surv_dat$censoringOS = 0
 surv_dat$censoringOS[surv_dat$Patientstatus == 'death by primary disease'] = 1
 
-#Censoring for disease free survival (1 = event (recurrence), 0 = censored)
+# Censoring for disease free survival (1 = event (recurrence), 0 = censored)
 surv_dat$censoringDFS = 0
 surv_dat$censoringDFS[surv_dat$Patientstatus == 'alive w metastases'] = 1
 surv_dat$censoringDFS[surv_dat$OSmonth > surv_dat$DFSmonth] = 1 
```

Comment:
Rather a general feedback, just picking this line as an example.
As you work with `data.table`s you could also rewrite these kind of statements a bit more concise/faster/idiomatic:
```
surv_dat[ , censoringDFS := 0]
surv_dat[ Patientstatus == 'alive w metastases' , censoringDFS := 1]
surv_dat[ OSmonth > DFSmonth, censoringDFS := 1]
```
As general data.table's in place assignment via `:=` is faster then modifying columns otherwise and also selection of rows via the data table synthax is faster.

This statement could be even further condensed into:
```
surv_dat[ , censoringDFS := as.numeric((Patientstatus == 'alive w metastases') | ( OSmonth > DFSmonth)]
```

Reply:
I agree that this will make it faster and the code will be more compact. But also think about the readability. Since these are examples that anyone should use, I'd actually recommend using base R implementations that are less efficient. Or at least one needs to explain in words what exactly happens in the call.
But happy to be overruled ;)

Reply:
In this case I would almost recommend coercing the data into a `data.frame` after reading data in or using base R to begin with (i.e. `read.csv` or so instead of the fast `fread`).

The reason is that `data.table`s and `data.frame`s do not always behave exactly identically and need sometimes slightly different syntax.  Working idiomatically makes it explicit to the reader that this is not a simple `data.frame` but a `data.table` and thus `data.table` rules apply, which I do prefer.

However, as mixing the two - in many cases - works fine, I don't feel to strongly about this.

Reply:
Ok, then it's better to go full `data.table`

Reply:
Sure, full data.table sounds good.

Reply:
I think I'd go with Vito's first version for readability though and not the super condensed version.